ProductionModePlugin.js

```diff
@@ -112,27 +74,33 @@ ProductionModePlugin.prototype.apply = function(compiler) {
 
     if (defaultMessages) {
       attributes.messages = attributes.messages ? objectDeepClone(attributes.messages) : {};
-      attributes.messages[locale] = attributes.messages[locale] || {};
-      extend(attributes.messages[locale], defaultMessages);
+      localeMessages = attributes.messages[locale] || {};
+      attributes.messages[locale] = merge(localeMessages, defaultMessages);
     }
-  });
 
-  /*
-  TODO
-  // Generate default translation.
-  globalizeCompiler.generateDefaultTranslation({
-    path,
-    defaultLocale,
-    defaultMessages
+    process.nextTick(function () {
+      writeMessages(locale, attributes.messages[locale]);
+    });
   });
 
-  // Initialize or update translation.
-  globalizeCompiler.initOrUpdateTranslation({
-    filepath,
-    locale
-    defaultMessages
-  });
-  */
+  function writeMessages(locale, messages) {
+    var path = attributes.messages.replace("[locale]", locale);
+    var extractDefault = attributes.extractDefaultMessages || attributes.extractAllMessages;
+
+    if (extractDefault && locale === defaultLocale) {
+      reactGlobalizeCompiler.generateTranslation({
+        defaultLocale: defaultLocale,
+        defaultMessages: messages,
+        filepath: path
+      });
+    } else if (attributes.extractAllMessages) {
+      reactGlobalizeCompiler.initOrUpdateTranslation({
+        defaultMessages: messages,
+        filepath: path,
+        locale: locale
+      });
+    }
```

Comment:
hm - I think you should be able to use the plugin if you don't want it to write any messages file.

Maybe add a check for `attributes.messages` in the plugin function - `compiler.plugin("globalize-before-compile-extracts"` and then warn? So if the user doesn't set `messages: true`, it just doesn't bother trying to write anything.